internal/pkg/agent/configuration/grpc.go

```diff
@@ -4,14 +4,17 @@
 
 package configuration
 
-import "fmt"
+import (
+	"fmt"
+)
 
 // GRPCConfig is a configuration of GRPC server.
 type GRPCConfig struct {
 	Address                 string `config:"address"`
 	Port                    uint16 `config:"port"`
 	MaxMsgSize              int    `config:"max_message_size"`
 	CheckinChunkingDisabled bool   `config:"checkin_chunking_disabled"`
+	Local                   bool   `config:"local"` // Enables RPC via domain socker/named pipe
```

Comment:
This really is an internal functionality and I would prefer to see it enabled by default. Would it be possible to just change the port default to a `int32` with `-1` as the default. That would allow anyone that has configured a different port to be used how it works today, and if its `-1`. Possibly the same with the address field. If its empty and its -1 for the port, it used the unix/npipe otherwise it fallsback to how it was configured.

If we hit an issue the solution is to set the `address` to `localhost` and `port` to `6789`.

Reply:
Are we absolutely positive that we want to break the default behavior?

I thought that it would be safer to keep default as IP socket: ```address``` to ```localhost``` and ```port``` to ```6789```
and the let the user to switch to domain socket/named pipe.

Changing ```Port``` to ```int16``` will limit the port number that users can use. Could potentially change it to ```int32```.

Introducing the separate/additional attribute seemed cleaner. 

Let me know if you still want to have -1 ```Port``` as default and remove ```Local``` attribute.

Reply:
I would prefer to see this change and become the default due to the SDH we get with using network based communication. I think being that this communication is internal its not a breaking change.

@cmacknz What do you think?

Reply:
Agree, we need to stop using localhost by default. The communication method is internal to the agent so I do not consider it a breaking change.

I do think it is important that we be able to switch from unix sockets/pipes back to localhost via configuration. This is how we can make the change safely and protect ourselves from unknown unknowns.

Reply:
Sounds good. Thanks for confirmation, will update PR then to use -1 for the port default (pipes rpc)  as suggested.

Reply:
Addressed this requirement now.